src/components/ResourceBrowser/ResourceList/ConnectingToClusterState.tsx

```diff
@@ -39,57 +58,34 @@ export default function ConnectingToClusterState({
         }
     }, [clusterId, selectedCluster])
 
-    const initProgressTimer = () => {
-        if (progressTimer) {
-            clearTimeout(progressTimer)
-        }
-
-        progressTimer = setTimeout(() => {
-            setInfoText(TAKING_LONGER_TO_CONNECT)
-            setShowCancel(true)
-
-            if (progressTimer) {
-                clearTimeout(progressTimer)
-            }
-        }, 10000)
-    }
-
-    const renderInfo = (heading: string, infoText: string) => {
+    const renderInfo = (heading: string, _infoText: string) => {
         return (
             <>
                 <h2 className="fs-16 fw-6 lh-24 mt-20 mb-8 w-300" data-testid="cluster_info_getting_loaded">
                     {heading}
                 </h2>
-                <p className="fs-13 fw-4 lh-20 w-300 mb-20">{infoText}</p>
+                <p className="fs-13 fw-4 lh-20 w-300 mb-20">{_infoText}</p>
             </>
         )
     }
 
-    const resetStates = () => {
-        setInfoText(TRYING_TO_CONNECT)
-        setShowCancel(false)
-        setErrorMsg('')
-        setResetProgress(!resetProgress)
-    }
-
     const handleCancelClick = () => {
-        sideDataAbortController.new.abort()
-        sideDataAbortController.prev = sideDataAbortController.new
+        requestAbortController?.abort()
         resetStates()
-        setSelectedCluster(null)
         replace({
             pathname: URLS.RESOURCE_BROWSER,
         })
     }
 
-    const handleRetryClick = (e) => {
+    const handleRetryClick: React.MouseEventHandler<HTMLButtonElement> = (event) => {
         resetStates()
         initProgressTimer()
-        handleRetry(e)
+        handleRetry(event)
     }
 
     const renderSelectionState = () => {
-        if (loader && !errorMsg) {
+        /* NOTE: should show loading irrespective of errorMsg */
+        if (loader) {
```

Comment:
In case of errorMessage it should show the error

Reply:
I am certain the loader will not keeping loading forever since it will be coming down from useAsync.